src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -122,17 +123,25 @@ private static ProxyGeneratorResult GetProxyResultForNoDefaultConstructor(Type t
         private static object DoGenerateProxy(
             Type typeOfProxy,
             ProxyGenerationOptions options,
-            IEnumerable<Type> additionalInterfacesToImplement,
+            ReadOnlyCollection<Type> additionalInterfacesToImplement,
             IEnumerable<object> argumentsForConstructor,
             IInterceptor interceptor)
         {
-            var allInterfacesToImplement = additionalInterfacesToImplement;
+            Type[] allInterfacesToImplement;
 
             if (typeOfProxy.GetTypeInfo().IsInterface)
             {
-                allInterfacesToImplement = new[] { typeOfProxy }.Concat(allInterfacesToImplement);
+                allInterfacesToImplement = new Type[1 + additionalInterfacesToImplement.Count];
+                additionalInterfacesToImplement.CopyTo(allInterfacesToImplement, 1);
+
+                allInterfacesToImplement[0] = typeOfProxy;
                 typeOfProxy = typeof(object);
             }
+            else
+            {
+                allInterfacesToImplement = new Type[additionalInterfacesToImplement.Count];
+                additionalInterfacesToImplement.CopyTo(allInterfacesToImplement, 0);

```

Comment:
Could you perhaps pass `additionalInterfacesToImplement` as an array rather than `ReadOnlyCollection<Type>`? It would avoid this copy.